### ISSUE TITLE
Move inline script to ps1 script

### ISSUE THREAD

**Issue by chcosta** [dotnet/arcade#6088]
Inlined scripts are difficult to debug and *usually* not necessary.  The inlined script in Arcade's [eng/promote-build.yml](
https://github.com/dotnet/arcade/blob/master/eng/promote-build.yml#L76-L130) should be moved to a checked in script.  This inlined script is also, now in windowsdesktop and will need to be updated there.

@epananth

**Comment by riarenas** [comment]
Not sure if that's the case for this one, but one reason why we moved heavily towards inline scripts for smaller ones during the 3.0 timeframe was build times, as repos need to be cloned and checked out for checked in scripts to run, and cloning the bigger repos was adding a lot of time to builds.

For v3 I think we can check in more scripts, but I think the rule should be something like:

* If the script runs during the main repo build, it should stay inline
* if the script is going to run during the build promotion pipeline, it can be checked in.

**Comment by chcosta** [comment]
How much extra time are we talking?  validating / investigating inline scripts is a pain.  I'll cede the point that these are not meant to be run locally, and it's not straight forward to run these outside of an AzDo environment, but it should be possible.  We should consider other options, like a light-weight [resource repository](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#resources) rather than inlining this stuff.  Meanwhile, I'm completely ok with this specific issue being closed, I just wanted to make sure the PR feedback in https://github.com/dotnet/windowsdesktop/pull/1069#pullrequestreview-479801701 was captured.

**Comment by riarenas** [comment]
Depending on the repo (mostly runtime), it was something like 3 minutes per each stage that had to checkout.

I don't disagree with this issue, and there's cleanup that we can do now that a lot of the publishing code is being moved out of the repo build. I'm mostly just giving context on why we ended up preferring inline scripts for steps that had to run multiple times during the publishing stages.

As general feedback, I don't think it's a good idea to copy the arcade yaml into particular repos, inline scripts or otherwise. It just smells of either a deficiency of v3 publishing, or a problem with how the repo is using the share infrastructure.

https://github.com/dotnet/arcade/pull/3999 had some info on this.

**Comment by chcosta** [comment]
> As general feedback, I don't think it's a good idea to copy the arcade yaml into particular repos, inline scripts or otherwise.

Completely agree.  That type of cleanup was the intent of https://github.com/dotnet/arcade/issues/6088#issuecomment-684999063, but I scoped the issue too much.  I've added another note based on this discussion.

**Comment by garath** [comment]
@mmitche Your thoughts on this?

**Comment by mmitche** [comment]
Yeah, that promote-build should live in arcade's common scripts and be updated there. I thought that was what we were going to do though? We have a top level hook into the post-build process for the repo (post-build.yml). Could we not have just conditionalized import of the various templates based on the publishingInfraVersion?
